unit_test/transforms/test_transform_utils.py

```diff
@@ -0,0 +1,200 @@
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import unittest
+from omotes_simulator_core.adapter.transforms.transform_utils import (
+    sort_ports,
+    order_prim_sec_ports,
+    reverse_dict,
+    PortType,
+    Port,
+)
+
+
+class TransformUtilsTest(unittest.TestCase):
+    """Class to test the transform utils functions."""
+
+    def setUp(self) -> None:
+        """Set up test case."""
+        self.connected_ports = [
+            Port(port_id="port1", port_name="Prim port1", port_type=PortType.IN),
+            Port(port_id="port2", port_name="Prim port2", port_type=PortType.OUT),
+        ]
+
+    def test_sort_ports_basic(self):
+        """Test the port sorting function. THe order is nto changed."""
+        # arrange
+
+        # act
+        result = sort_ports(self.connected_ports)
+
+        # assert
+        self.assertEqual(result, [port.port_id for port in self.connected_ports])
+
+    def test_sort_ports_reverse(self):
+        """Test the port sorting function. The order is reversed."""
+        # arrange
+        connected_ports = self.connected_ports[::-1]
+
+        # act
+        result = sort_ports(connected_ports)
+
+        # assert
+        self.assertEqual(result, [port.port_id for port in self.connected_ports])
+
+    def test_sort_ports_4_ports(self):
+        """Test the port sorting function with 4 ports, the order remains the same."""
+        # arrange
+        self.connected_ports.append(
+            Port(port_id="port3", port_name="Sec port3", port_type=PortType.IN)
+        )
+        self.connected_ports.append(
+            Port(port_id="port4", port_name="Sec port4", port_type=PortType.OUT)
+        )
+        # act
+        result = sort_ports(self.connected_ports)
+
+        # assert
+        self.assertEqual(result, [port.port_id for port in self.connected_ports])
+
+    def test_sort_ports_4_ports_reverse(self):
+        """Test the port sorting function with 4 ports, the order is reversed."""
+        # arrange
+        self.connected_ports.append(
+            Port(port_id="port3", port_name="Sec port3", port_type=PortType.IN)
+        )
+        self.connected_ports.append(
+            Port(port_id="port4", port_name="Sec port4", port_type=PortType.OUT)
+        )
+        connected_ports = self.connected_ports[::-1]
+
+        # act
+        result = sort_ports(connected_ports)
+
+        # assert
+        self.assertEqual(result, [port.port_id for port in self.connected_ports])
+
+    def test_sort_port_error_number_of_ports(self):
+        """Test the port sorting function.
+
+        This test has different number of in and out port resulting in error.
+        """
+        # arrange
+        self.connected_ports.append(
+            Port(port_id="port3", port_name="Sec port3", port_type=PortType.IN)
+        )
+
+        # act
+        with self.assertRaises(ValueError) as cm:
```

Comment:
Why should this fail?

Reply:
Since the number of in and outports is not equal.

Reply:
I was considering whether this is a must. I can imagine - don't ask why - that someone wants to model 'multiple' in-ports. 

The check is fine as long as we somehow document that you need equal in and out ports.